keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/deliveryconfigs/deliveryconfigs.kt

```diff
@@ -34,7 +36,7 @@ internal fun SqlStorageContext.deliveryConfigByName(name: String): DeliveryConfi
           name = name,
           application = application,
           serviceAccount = serviceAccount,
-          metadata = metadata as Map<String, Any?>? ?: emptyMap()
+          metadata = (metadata ?: emptyMap()) + mapOf("createdAt" to ULID.parseULID(uid).timestampAsInstant())
```

Comment:
Is this to cater for existing records where `createdAt` wasn't persisted? If we're doing this on read, why bother to persist it at all?

Reply:
We're not persisting it. It's extracted from the ULID... The only reason I added this was so that I could have something to compare against in the tests, since `uid` is not exposed in `DeliveryConfig` and I didn't want to add `createdAt` as an extra field in there either.

Reply:
I could expose the method to read the UID instead in the interface if you think that's preferable.

Reply:
P.S. I should have added this same comment here: https://github.com/spinnaker/keel/pull/1833/files#r585070191